kyo-core/shared/src/test/scala/kyo/FiberTest.scala

```diff
@@ -137,25 +137,21 @@ class FiberTest extends Test:
         "interrupts losers" - {
             "promise + plain value" in runJVM {
                 for
-                    flag        <- AtomicBoolean.init(false)
-                    promise     <- Promise.init[Nothing, Int]
-                    _           <- promise.onInterrupt(_ => flag.set(true))
-                    result      <- Async.race(promise.get, 42)
-                    interrupted <- flag.get
```

Comment:
There was no guarantee that the atomic set by the `onInterrupt` function is already executed before the flag is read. The fix uses a latch to park execution and wait for the `onInterrupt` function to execute.